### ISSUE TITLE
How to understand and use focalDistance in OrthographicCamera and PerspectiveCamera?

### ISSUE THREAD

**Issue by kanition** [mmp/pbrt-v4#232]
Hi~ I really love this impressive work! I'd like to ask a question as the title says: 
### How to understand and use focalDistance in OrthographicCamera and PerspectiveCamera?

I find the definetion and usage of "focalDistance" may be conflict if I don't misunderstand it.

- The book [here](https://www.pbr-book.org/3ed-2018/Camera_Models/Projective_Camera_Models#eq:thin-lens-focus) says that "focalDistance" is defined as
> the distance between the lens and the film that sets the plane of focus at some z.

- But it seems to work in another way in the class OrthographicCamera:
https://github.com/mmp/pbrt-v4/blob/56c0e6db86f9bee2d2fd1ac7c879eddfb161422c/src/pbrt/cameras.cpp#L287-L303
When `lensRadius > 0` is true, it can be showed that `pFocus`=`pCamera`+`(0, 0, focalDistance)`,
which means that `focalDistance` is the distance along z axis between the near plane (i.e. the film plane but in front of the lens) and the plane of focus, rather than "the distance between the lens and the film", as `pLens` is `(pLens.x, pLens.y, 0)` so it should be `pCamera.z`.

- The situation changes in the class PerspectiveCamera:
https://github.com/mmp/pbrt-v4/blob/56c0e6db86f9bee2d2fd1ac7c879eddfb161422c/src/pbrt/cameras.cpp#L402-L419
When `lensRadius > 0` is true, initially the origin of `ray` is `(0, 0, 0)`, so it is on the lens plane as `pLens` is `(pLens.x, pLens.y, 0)`.
Then it can be showed that `pFocus`=`ft` * `Normalize(Vector3f(pCamera))`. So `pFocus.z` is `focalDistance`.
It means that "focalDistance" is the distance between the lens plane and the plane of focus, which is different from the both introduced above.

- In conclusion, there are three types of "focalDistance":
1. the lens <-> the film
2. the near plane <-> the plane of focus
3. the lens <-> the plane of focus

How to understand them? Thank you!

**Comment by pierremoreau** [comment]
1. and 3. will define the same distance, as the film is located in the focal plane IIRC. There are two focal planes located on each side of the lens and at equal distance (the focal length) of it.

Regarding 2., I am not sure. If the near plane is fairly close to the lens, the distance between the near plane and the focal plane would be almost the same as the distance between the lens and the focal plane, i.e. the focal length.
